dispatch/src/dispatch.rs

```diff
@@ -401,7 +395,14 @@ impl<'pipe> ApduDispatch<'pipe> {
                 _ => panic!("Unexpected buffer state."),
             };
 
-            self.current_aid = Some(aid);
+            let old_aid = mem::replace(&mut self.current_aid, Some(aid));
```

Comment:
nit: Can we use [`Option::replace`](https://doc.rust-lang.org/std/option/enum.Option.html#method.replace) instead?  IMHO easier to read.

```suggestion
            let old_aid = self.current_aid.replace(aid);
```

Reply:
Good catch. Probably should be [a clippy lint](https://github.com/rust-lang/rust-clippy/issues/14195).